src/mui/input/ReferenceArrayInput.js

```diff
@@ -232,44 +301,45 @@ ReferenceArrayInput.propTypes = {
         order: PropTypes.oneOf(['ASC', 'DESC']),
     }),
     source: PropTypes.string,
+    translate: PropTypes.func.isRequired,
 };
 
 ReferenceArrayInput.defaultProps = {
     allowEmpty: false,
     filter: {},
     filterToQuery: searchText => ({ q: searchText }),
-    matchingReferences: [],
+    matchingReferences: null,
     perPage: 25,
-    sort: { field: 'id', order: 'DESC' },
     referenceRecords: [],
+    sort: { field: 'id', order: 'DESC' },
 };
 
 function mapStateToProps(state, props) {
     const referenceIds = props.input.value || [];
     const data = state.admin.resources[props.reference].data;
     return {
-        referenceRecords: referenceIds.reduce((references, referenceId) => {
-            if (data[referenceId]) {
-                references.push(data[referenceId]);
-            }
-            return references;
-        }, []),
         matchingReferences: getPossibleReferences(
             state,
             referenceSource(props.resource, props.source),
             props.reference,
             referenceIds
         ),
+        referenceRecords: referenceIds.reduce((references, referenceId) => {
+            if (data[referenceId]) {
+                references.push(data[referenceId]);
+            }
+            return references;
+        }, []),
     };
 }
 
-const ConnectedReferenceInput = connect(mapStateToProps, {
+const ConnectedReferenceArrayInput = connect(mapStateToProps, {
     crudGetMany: crudGetManyAction,
     crudGetMatching: crudGetMatchingAction,
 })(ReferenceArrayInput);
 
-ConnectedReferenceInput.defaultProps = {
+ConnectedReferenceArrayInput.defaultProps = {
     addField: true,
 };
 
-export default ConnectedReferenceInput;
+export default translate(ConnectedReferenceArrayInput);
```

Comment:
Why do you decorate with translate here and not line 336?

Reply:
After review, it's working.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/547706/36200128-f1da53c8-117b-11e8-910f-a1d1f0226ed2.gif)

![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/547706/36200171-18b85864-117c-11e8-80c7-ef580cfab753.gif)